language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/SourceGen.java

```diff
@@ -2016,6 +2123,210 @@ public String getSourceForFunction(JsonObject node, boolean pretty, boolean repl
                  + join(node.getAsJsonArray("workers"), pretty, replaceLambda, "", null, false, sourceGenParams)
                  + outdent(node, sourceGenParams.isShouldIndent()) + w("", sourceGenParams) + "}"
                  + a("", sourceGenParams.isShouldIndent());
+            } else if (node.get("native") != null && node.get("native") .getAsBoolean()
+                         && node.get("markdownDocumentationAttachment") != null
+                         && node.get("annotationAttachments") != null
+                         && node.get("deprecatedAttachments") != null
+                         && node.getAsJsonObject("name").get("valueWithBar") != null
+                         && !node.getAsJsonObject("name").get("valueWithBar").getAsString().isEmpty() && node.get("allParams") != null
+                         && node.get("restParameters") != null && node.get("hasReturns") != null
+                         && node.get("hasReturns") .getAsBoolean()
+                         && node.get("returnTypeAnnotationAttachments") != null
+                         && node.get("returnTypeNode") != null) {
```

Comment:
if we are matching specific scenarios (eg. a native function) in these if conditions; shall we move those into private methods for the readability. For example; then we may write `if (isNativeFunction(...)) {`

Reply:
this is a generated class. It generating this code from a grammar file (tree.g). It is bit hard to isolate these as the generator is already a bit complex... i'm thinking of a refactor to make it bit simple and will consider this generation when i start that.